mapiproxy/servers/default/emsmdb/emsmdbp_provisioning.c

```diff
@@ -552,15 +550,25 @@ FolderId: 0x67ca828f02000001      Display Name: "                        ";  Con
 				mapistore_url = talloc_asprintf(mem_ctx, "%s0x%"PRIx64"/", fallback_url, current_fid);
 			}
 
-			/* Ensure the name is unique */
-			base_name = current_name;
-			j = 1;
-			while (openchangedb_get_fid_by_name(emsmdbp_ctx->oc_ctx, username, ipm_fid, current_name, &found_fid) == MAPI_E_SUCCESS) {
-				current_name = talloc_asprintf(mem_ctx, "%s (%d)", base_name, j);
-				j++;
+			/* According to [MS-OXOSFLD] Section 3.1.4.1, the special folder must be reused
+			   if one already exists by that name */
+			ret = openchangedb_get_fid_by_name(emsmdbp_ctx->oc_ctx, username, ipm_fid, current_name, &found_fid);
+			if (ret == MAPI_E_NOT_FOUND) {
+				mapistore_indexing_get_new_folderID_as_user(emsmdbp_ctx->mstore_ctx, username, &current_fid);
+				openchangedb_get_new_changeNumber(emsmdbp_ctx->oc_ctx, username, &current_cn);
+				openchangedb_create_folder(emsmdbp_ctx->oc_ctx, username, ipm_fid, current_fid, current_cn, mapistore_url, i);
+			} else if (ret == MAPI_E_SUCCESS) {
+				current_fid = found_fid;
+				openchangedb_set_mapistoreURI(emsmdbp_ctx->oc_ctx, username, current_fid, mapistore_url);
+				openchangedb_set_system_idx(emsmdbp_ctx->oc_ctx, username, current_fid, i);
+
+				/* change number is updated by set_folder_properties */
+			} else {
+				OC_DEBUG(1, "Error trying to get fid by name (%s): %s", current_name, mapi_get_errstr(ret));
+				/* Should we halt the process? */
+				continue;
```

Comment:
I'd put `goto error;`. Probably this will never happen but if it does there is a mayor error so no need to move forward in this  function